File: src/rust/cat.rs
/* Light builtins's (Rust version)
 * cat - concatenate files and print on the standard output
 * A lightweight repository of useful shell commands
 * GitHub: https://www.github.com/lewisevans2007/light_builtins
 * Licence: GNU General Public License v3.0
 * By: Lewis Evans
*/

use std::env;
use std::fs;

fn main() {
    let args: Vec<String> = env::args().collect();
    /* Get the command line arguments */
    let path;
    if args.len() == 1 {
        /* Check if a file path was provided */
        /* Print an error message and exit the program */
        println!("cat: missing file operand");
        return;
    } else {
        /* Get the file path from the command line arguments */
        path = &args[1];
    }
    /* Read the file contents */
    let content = match fs::read_to_string(path) {
        Result::Ok(value) => value,
        Result::Err(_error) => {
            println!("cat: {}: No such file or directory", path);
            return;
        }
    };

    println!("{}", content);
}

File: src/python/cat.py
# Light builtins's (Python version)
# cat - concatenate files and print on the standard output
# A lightweight repository of useful shell commands
# GitHub: https://www.github.com/lewisevans2007/light_builtins
# Licence: GNU General Public License v3.0
# By: Lewis Evans

import os
import sys

def main(args):
    if len(args) == 0:
        print("cat: missing operand")
        return 1
    for arg in args:
        if os.path.isfile(arg):
            with open(arg, "r") as file:
                print(file.read())
        else:
            print("cat: " + arg + ": No such file or directory")
            return 1
    return 0

if __name__ == "__main__":
    sys.exit(main(sys.argv[1:]))